pymc_experimental/inference/pathfinder.py

```diff
@@ -0,0 +1,138 @@
+#   Copyright 2022 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import collections
+import sys
+
+import arviz as az
+import blackjax
+import jax
+import jax.numpy as jnp
+import jax.random as random
+import numpy as np
+import pymc as pm
+from pymc import modelcontext
+from pymc.sampling_jax import get_jaxified_graph
+from pymc.util import get_default_varnames
+
+
+def convert_flat_trace_to_idata(
+    samples,
+    dims=None,
+    coords=None,
+    include_transformed=False,
+    postprocessing_backend="cpu",
+    model=None,
+):
+    model = modelcontext(model)
+    init_position_dict = model.initial_point()
+    trace = collections.defaultdict(list)
+    astart = pm.blocking.DictToArrayBijection.map(init_position_dict)
+    for sample in samples:
+        raveld_vars = pm.blocking.RaveledVars(sample, astart.point_map_info)
+        point = pm.blocking.DictToArrayBijection.rmap(raveld_vars, init_position_dict)
+        for p, v in point.items():
+            trace[p].append(v.tolist())
+
+    trace = {k: np.asarray(v)[None, ...] for k, v in trace.items()}
+
+    var_names = model.unobserved_value_vars
+    vars_to_sample = list(get_default_varnames(var_names, include_transformed=include_transformed))
+    print("Transforming variables...", file=sys.stdout)
+    jax_fn = get_jaxified_graph(inputs=model.value_vars, outputs=vars_to_sample)
+    result = jax.vmap(jax.vmap(jax_fn))(
+        *jax.device_put(list(trace.values()), jax.devices(postprocessing_backend)[0])
+    )
+
+    trace = {v.name: r for v, r in zip(vars_to_sample, result)}
+    idata = az.from_dict(trace, dims=dims, coords=coords)
+
+    return idata
+
+
+def fit_pathfinder(
+    iterations=5_000, random_seed=312, postprocessing_backend="cpu", ftol=1e-4, model=None
```

Comment:
```suggestion
    iterations=5_000, random_seed: Optional[RandomSeed] = None, postprocessing_backend="cpu", ftol=1e-4, model=None
```
We then usually convert the random_seed via 
```python
(random_seed,) = _get_seeds_per_chain(random_seed, 1)
```
That will follow the API for all other sampling methods, and accept integer/RandomState/RandomGenerator